docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc

```diff
@@ -0,0 +1,196 @@
+[[hints-annotations-autodiscovery]]
+= Hints annotations based autodiscover
+
+beta[]
+
+NOTE: Make sure you are using {agent} 8.5+.
+
+Standalone {agent} supports autodiscover based on hints from the <<kubernetes-provider,provider>>.
+The hints mechanism looks for hints in Kubernetes Pod annotations that have the prefix `co.elastic.hints`.
+As soon as the container starts, {agent} checks it for hints and launches the proper configuration
+for the container. Hints tell {agent} how to monitor the container by using the proper integration.
+This is the full list of supported hints:
+
+[discrete]
+== Required hints:
+
+[float]
+=== `co.elastic.hints/package`
+
+The package to use for monitoring.
+
+[float]
+=== `co.elastic.hints/host`
+
+The host to use for metrics retrieval.
+
+[discrete]
+== Optional hints available:
+
+[float]
+=== `co.elastic.hints/data_stream`
+
+The list of data streams to enable. If not specified, the integration's default data streams are used.
+To find the defaults, refer to the {integrations-docs}[Elastic integrations documentation].
+
+If data streams are specified, hints can be defined for each one. For example, `co.elastic.hints/key.period: 10m` if the data stream specified is `info` from the {metricbeat-ref}/metricbeat-module-redis.html[Redis module]. Otherwise, the data stream will use the top-level hints
```

Comment:
```suggestion
If data streams are specified, additional hints can be defined per data stream. 

For example, `co.elastic.hints/info.period: 5m` if the data stream specified is `info` for the {metricbeat-ref}/metricbeat-module-redis.html[Redis module]. 

[source,yaml]
----
apiVersion: v1
kind: Pod
metadata:
  name: redis
  annotations:
    co.elastic.hints/package: redis
    co.elastic.hints/data_streams: info
    co.elastic.hints/info.period: 5m
----

```
Otherwise, the data stream will use the top-level hints
```

Reply:
@ChrsMark is info.period not key.period right?